asllib/ASLTypingReference.mld

```diff
@@ -722,124 +722,261 @@ of the following applies:
 
   {3 TypingRule.SCond}
 
     {4 Rule}
+    All of the following applies:
+    - [s] is a condition [e] with two statements [s1] and [s2];
+    - [t_cond, e_cond] is the result of annotating [e] in [env];
+    - [t_cond] type-satisfies [t_bool];
+    - [s1'] is the result of annotating [s1] in [env];
+    - [s2'] is the result of annotating [s2] in [env];
+    - [new_s] is the condition [e_cond] with two statements [s1'] and [s2'];
+    - [new_env] is [env].
 
     {4 Example}
 
   {3 TypingRule.SCase}
 
     {4 Rule}
-    <description>
+    All of the following applies:
+    - [s] is a case statement with expression [e] and cases [cases];
+    - [t_e, e1] is the result of annotating [e] in [env];
+    - [cases1, env1] is the result of annotating each case in [cases];
+    - [new_s] is a case statement with expression [e1] and cases [cases1];
+    - [new_env] is [env1].
 
     {4 Example}
 
   {3 TypingRule.SAssert}
 
     {4 Rule}
+    All of the following applies:
+    - [s] is an assert statement with expression [e];
+    - [t_e',e'] is the result of annotating [e] in [env];
+    - [t_e'] type-satisfies [t_bool];  
+    - [new_s] is an assert statement with expression [e'];
+    - [new_env] is [env].
 
     {4 Example}
 
   {3 TypingRule.SWhile}
 
     {4 Rule}
+    All of the following applies:
+    - [s] is a while statement with expression [e1] and statement [s1];
+    - [t, e2] is the result of annotating [e1] in [env];
+    - [t] type-satisfies [t_bool];
+    - [s2] is the result of annotating [s1] in [env];
+    - [new_s] is a while statement with expression [e2] and statement [s2];
+    - [new_env] is [env].
 
     {4 Example}
 
   {3 TypingRule.SRepeat}
 
     {4 Rule}
+    All of the following applies:
+    - [s] is a repeat statement with expression [e1] and statement [s1];
+    - [s2] is the result of annotating [s1] in [env];
+    - [t, e2] is the result of annotating [e1] in [env];
+    - [t] type-satisfies [t_bool];
+    - [new_s] is a repeat statement with expression [e2] and statement [s2];
+    - [new_env] is [env].
 
     {4 Example}
 
   {3 TypingRule.SFor}
 
     {4 Rule}
+    All of the following applies:
+    - [s] is a for statement with index [id], direction [dir], two expressions
+      [e1] and [e2] and a statement [s'];
+    - [t1,e1'] is the result of annotating [e1] in [env];
+    - [t2,e2'] is the result of annotating [e2] in [env];
+    - [t1] has the structure of an integer type;
+    - [t2] has the structure of an integer type;
+    - [ty] is ;
+    - [s''] is the result of annotating [s'] in [env];
+    - [new_s] is a for statement with index [id], direction [dir], two expressions
+      [e1'] and [e2'] and statement [s''];
+    - [new_env] is [env].
 
     {4 Example}
 
-  {3 TypingRule.SThrowNone}
+  {3 TypingRule.SDeclSome}
 
     {4 Rule}
+    All of the following applies:
+    - [s] is a declaration with [ldk], [ldi] and an optional expression [e_opt];
+    - [e_opt] gives some expression [e];
```

Comment:
I think this is equivalent and more explicit (we have a degree less of complexity)
```suggestion
    - [s] is a declaration with [ldk], [ldi] and an expression [e];
```